spec/policies/invitation_policy_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe InvitationPolicy do
+  subject { described_class }
+
+  let(:member) { build(:user) }
+  let(:admin) { build(:user, :admin) }
+
+  permissions :new?, :create? do
+    it 'denies access to regular member users' do
+      expect(subject).not_to permit(member)
+    end
+
+    it 'grants access to admin users' do
+      expect(subject).to permit(admin)
+    end
+  end
+
+  permissions :edit?, :update? do
```

Comment:
What would be a use case for this? Also shouldn't it be scoped to that user directly can only edit/update their own?

Reply:
This is kind of due to the way the gem works, I was initially a bit puzzled by it.

If you look at `app/views/devise/invitations/edit.html.erb` you'll see what essentially looks like a sign up page. What happens is a user is created with their email address when the invite is initially sent. When the user clicks on the invite link in their email, they're taken to that "edit" page to enter their password (and potentially `timezone`, etc. in the future) and of course `update` is what happens when they submit.

This flow is already fully locked down by the invitation token. No user can visit the edit page or submit a response to it without a valid invitation token. So I just put `true` there to keep pundit out of the way on that. It's already tightly controlled by the `devise_invitations` gem itself.

Reply:
ahhh perfect. Thank you for this explanation.

Reply:
Cool, I'll have this flow  in mind when implementing timezone